### PR TITLE
Clean up zorder .g4

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Clean up .g4 file. Remove some grammar we donn't need

### _How was this patch tested?_
Pass CI

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @qphien @cfmcgrady @yaooqinn @pan3793

**Comment by yaooqinn** [comment]
can you add it as a sub-task of #939?

**Review by yaooqinn — APPROVED** [review]

File: dev/kyuubi-extension-spark-3-1/pom.xml [hunk 1]
```diff
@@ -118,7 +118,7 @@
                 <artifactId>antlr4-maven-plugin</artifactId>
                 <configuration>
                     <visitor>true</visitor>
-                    <sourceDirectory>${project.basedir}/src/main/antlr4/org/apache/kyuubi/zorder</sourceDirectory>
+                    <sourceDirectory>../kyuubi-extension-spark-3-1/src/main/antlr4</sourceDirectory>
```
Comment (pan3793) [inline dev/kyuubi-extension-spark-3-1/pom.xml:121]: Can we keep `${project.basedir}` here?

**Review by pan3793 — APPROVED** [review]

File: dev/kyuubi-extension-spark-3-1/src/main/antlr4/org/apache/kyuubi/sql/zorder/ZorderSqlExtensions.g4 [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    : '('? multipartIdentifier comparisonOperator constant ')'?
+    ;
+
+comparisonOperator
+    : EQ | NEQ | NEQJ | LT | LTE | GT | GTE | NSEQ
+    ;
+
+constant
+    : NULL                                                            #nullLiteral
+    | identifier STRING                                               #typeConstructor
```
Comment (pan3793) [inline dev/kyuubi-extension-spark-3-1/src/main/antlr4/org/apache/kyuubi/sql/zorder/ZorderSqlExtensions.g4:81]: nit: can we shrink spaces here?

Comment (ulysses-you) [reply dev/kyuubi-extension-spark-3-1/src/main/antlr4/org/apache/kyuubi/sql/zorder/ZorderSqlExtensions.g4]: all changed to eight indention

**Review by cfmcgrady — APPROVED** [review]

**Comment by qphien** [comment]
LGTM.

**Comment by ulysses-you** [comment]
thanks for review, merging to master

**Pull request closed without merging**